DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/hardware_reuse/README.md

```diff
@@ -0,0 +1,358 @@
+
+## Key Implementation Details
+
+This sample illustrates some key concepts:
+- Simple hardware reuse by invoking the task function in a 'for' loop.
+- More flexible hardware reuse by invoking the function task using a task sequence wrapper object.
+
+### No resource sharing
+This tutorial demonstrates a design that computes the 'square root of a dot product' multiple times. This operation is contained within a function named `OpSqrt()`. Without resource sharing, multiple identical compute blocks for `OpSqrt()` are generated on the FPGA. 
+
+```c++
+D3Vector new_item, item;
+
+item = InputPipeA::read();
+new_item[0] = OpSqrt(item, coef1);
+item = InputPipeA::read();
+new_item[1] = OpSqrt(item, coef1);
+item = InputPipeA::read();
+new_item[2] = OpSqrt(item, coef1);
+
+OutputPipeZ::write(OpSqrt(new_item, coef2));
+```
+
+<img src="assets/hardware_reuse_naive.svg" />
+
+By sharing the compute block using a loop or task sequence, we can tell the compiler to re-use the compute block, and avoid replicating the compute block hardware on the FPGA. The important trade-off made by resource sharing is that multiple invocations of the same compute block will block each other, so the II of the loop in which the shared compute block is called will increase.
+
+### Resource sharing with a loop
+Instead of invoking the `OpSqrt()` function one by one, you may re-use the function by calling it in a loop. In the `loop` directory, the device code calls the `OpSqrt()` function in a loop.
+
+```c++
+struct VectorOp{
+  int len;
+
+  void operator()() const{
+    constexpr D3Vector coef1 = {0.2, 0.3, 0.4};
+    constexpr D3Vector coef2 = {0.6, 0.7, 0.8};
+
+    D3Vector new_item;
+
+    // Calling OpSqrt() in a loop will re-use it
+    for (int i = 0; i < 3; i++){
+      D3Vector item = InputPipeA::read();
+      new_item[i] = OpSqrt(item, coef1);
+    }
+
+    // Another square root block 
+    // will be generated for this function call
+    OutputPipeZ::write(OpSqrt(new_item, coef2));
+  }
+};
+```
+
+<img src="assets/hardware_reuse_loop.svg" />
+
+This is a simple, but effective way to share a resource, but it is not always convenient to put _all_ instances of a function call into the same loop. consider the fourth call to `OpSqrt()`, which depends on the outputs of the first three calls. We could write the complex multiplexer logic to select whether the `OpSqrt()` function should be processing the outputs from the reads from `InputPipeA` or combining the outputs of the first three calls, but this code would be hard to read and debug. A better solution is to use task sequences to let the compiler generate this logic for us.
```

Comment:
This "what"?

Avoid using just "this". It can often be unclear or ambiguous as to what the "this" is referring to.

This code? This example? This class? This technique?